homeworks/veizerattila/5_oop/car_fleet_mgr.py

```diff
@@ -0,0 +1,116 @@
+
+- Az osztály tartalmazza a következő metódusokat:
+OK      ● Egy konstruktor, amely beállítja a fenti attribútumokat.
+OK      ● Egy drive() metódus, amely adott számú kilométerrel növeli a kilométeróra állását, és csökkenti az üzemanyag-szintet (tételezzük fel, hogy 0.1% üzemanyag fogy megtett kilométerenként). A drive() metódus csak annyit km-et engedjen vezetni, amennyire elegendő üzemanyag van.
+OK      ● Egy refuel() metódus, amely feltölti az üzemanyag-szintet egy adott mennyiséggel. Figyelj a limitekre.
+
+OK  - Készíts egy Fleet osztályt, amely kezeli a Car objektumokat:
+OK      ● Az osztály rendelkezzen egy listával, amelyben az autók találhatóak.
+OK      ● Tartalmazzon metódusokat Car objektumok hozzáadására és eltávolítására a flottába/flottából.
+OK      ● Tartalmazzon egy metódust, amely összesíti a flotta összes autójának összes kilométerét.
+OK      ● Hozz létre néhány Car objektumot, add hozzá őket a flottához, hajts végre néhány műveletet (vezetés, tankolás), jelenítsd meg az autók állapotát és a flotta összesítő adatait."""
+
+##############################################################################
+######################## I. Osztályok létrehozása:############################
+##############################################################################
+
+
+######################## "Car" osztály létrehozása:###########################
+class Car:
+    # Konstruktor létrezhozása, ami beállítja a kért attribútomokat (márka, modell, év + km óra, benzin szint):
+    def __init__(self, brand: str, model: str, year: int):
+        self.brand = brand
+        self.model = model
+        self.year = year
+        self.mileage = 0                    # km
+        self.fuel_level = 100               # %
+
+    # Drive() metódus definiálása:
+    def drive(self, kilometers):
+        fuel_needed = kilometers * .1
+        if self.fuel_level > fuel_needed:
+            self.mileage += kilometers
+            self.fuel_level -= fuel_needed
+        else:
+            print(f'Nincs elég benzin.')
+
+    # Refuel() metódus definiálása:
+    def refuel(self, liter):
+        if self.fuel_level + liter > 100:   # ezt itt úgy kezelem le, hogy ha több lenne, mint a max. 100 l, akkor csak addig lehessen tölteni
+            self.fuel_level = 100
+        else:
+            self.fuel_level += liter
+
+    # A __str__ metódus behozása, "olvashatóan" formázott string visszadásához a print-nél:
+    def __str__(self):
+        return(f"Autó: {self.brand} {self.model} ({self.year})  |  Megtett út: {self.mileage} km  |  Üzemanyagszint (beleszámítva a tankolást is, ha volt): {self.fuel_level} %")
+
+
+######################## "Fleet" osztály létrehozása:#########################
+class Fleet:
+    # Konstruktor létrezhozása, a flotta autóinak listája:
+    def __init__(self):
+        self.cars = []
+    
+    # autó hozzáadása metódus definiálása:
+    def add_cars(self, car):
```

Comment:
Az add_cars() és a remove_cars() elnevezések egy picit félrevezetőek, hiszen mindkettő esetben 1-1 autót kezelsz.
Leíróbb lenne, ha egyes számot használnál: add_car(), remove_car().
ill. figyelj kérlek hogy ha használsz type annotation-öket, akkor használd őket minden method-ban. Pl lent a remove_cars()-ban ott van hogy a car egy Car típus kell legyen, de itt nincs.

Reply:
Ezt javítottam, ill a type annotation-t is.